zerver/lib/actions.py

```diff
@@ -5093,9 +5093,16 @@ def do_get_user_invites(user_profile: UserProfile) -> List[Dict[str, Any]]:
     active_value = getattr(confirmation_settings, 'STATUS_ACTIVE', 1)
 
     lowest_datetime = timezone_now() - datetime.timedelta(days=days_to_activate)
-    prereg_users = PreregistrationUser.objects.exclude(status=active_value).filter(
-        invited_at__gte=lowest_datetime,
-        referred_by__realm=user_profile.realm)
+    prereg_users = []
+    if user_profile.is_realm_admin:
+        prereg_users = PreregistrationUser.objects.exclude(status=active_value).filter(
+            invited_at__gte=lowest_datetime,
+            referred_by__realm=user_profile.realm)
+    else:
+        prereg_users = PreregistrationUser.objects.exclude(status=active_value).filter(
+            invited_at__gte=lowest_datetime,
+            referred_by__realm=user_profile.realm,
```

Comment:
This middle line does nothing.  I think you could make this block cleaner by doing 
`base_query = PreregistrationUser.objects.exclude(status=active_value).filter(invited_at__gte=lowest_datetime)`

And then the if/else` would set e.g. `prereg_users = base_query.filter(referred_by=user_profile.realm)`

(I think you also don't need the `prereg_users = []` block at the top)

Reply:
The updated version of this is wrong -- we still need to filter on realm for administrations.  Otherwise, on environments hosting more than one realm, we'd leak the data for other organizations!